cmd/snap/cmd_debug_disks.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/snapcore/snapd/gadget"
+)
+
+type cmdDiskMapping struct {
+	clientMixin
+}
+
+func init() {
+	cmd := addDebugCommand("disks",
```

Comment:
One more thing that would probably be useful is a debug command to print out the disk layout according to the gadget, eg `snap debug show-layout /snap/my-gadget/current` and then one could compare if things match up.

Reply:
yes I have that debug command too in a different branch waiting on some pre-reqs first